code/modules/mob/living/silicon/mommi/inventory.dm

```diff
@@ -166,17 +166,16 @@
 /mob/living/silicon/robot/mommi/proc/unequip_head()
 	// If there is a hat on the MoMMI's head
 	if(head_state)
-		// Select the MoMMI's claw
-		select_module(INV_SLOT_TOOL)
-
-		// Put the hat in the MoMMI's claw
-		put_in_active_hand(head_state)
-
 		// Remove the head_state icon from the client's screen
 		if (client)
 			client.screen -= head_state
 
-		// Delete the hat from the head
+		head_state.forceMove(loc)
+		head_state.dropped(src)
```

Comment:
MoMMI inventory code is ASS
ALL
OF
IT

I'm serious, check the rest of the file, drop_item doesn't even support anything but the held item